### ISSUE TITLE
Period flattening fails for a DASH manifest

### ISSUE THREAD

**Issue by ismena** [google/shaka-player#2537]
Reproducible on the current nightly build.

Repro:
In the console, run the code to get a LIVE manifest with Server Side ads from IMA SDK:

```js
var adM = shakaDemoMain.player_.getAdManager();
var video = shakaDemoMain.video_;
var adContainer = video.ui.getControls().getControlsContainer();
adM.initServerSide(adContainer, video);
var streamRequest = new google.ima.dai.api.LiveStreamRequest();
streamRequest.assetKey = '0ndl1dJcRmKDUPxTRjvdog';
adM.requestServerSideStream(streamRequest).then((uri) => shakaDemoMain.player_.load(uri));
```

Expected:
Manifest is parsed and played.

Obvesrved:
A "MANIFEST.PERIOD_FLATTENING_FAILED" error is thrown.

**Comment by joeyparrish** [comment]
Here's a screenshot of what I'm seeing in the debugger for this content.  It's audio-only, and the first period has 6 streams, all of which have the same MIME type and codecs.

![Issue 2537 - screenshot](https://user-images.githubusercontent.com/6843156/80761179-ee1dc780-8aee-11ea-8703-78916c25cbbd.png)

Each stream has an identical twin.  Two each of 44kHz stereo @95kbps, 44kHz stereo @127kbps, and 48kHz mono @54kbps.  All of the languages are set to "und", and none have any roles defined.  There is no way to differentiate between these pairs of input streams except by ID.

**Comment by joeyparrish** [comment]
I'm handing this off to @ismena.  My first thought was to try using the stream ID in the "best match" algorithm, as a way to differentiate between these strange twin streams and make sure that every streams gets used.  It might not be enough once we start looking across periods, but it should make `unusedStreamsPerPeriod[0]` into an empty set at the very least.  The algorithm will still fail if there are any unused input streams.

A fallback position could be to use `shaka.log.alwaysError` or `shaka.log.alwaysWarn` to log a message instead of failing in this case.  It should still be possible to play this content even without using all the twin inputs.  Throwing the error was a way to be extremely strict to make sure that we didn't miss something important, but for really pathologically strange content like this, it could always be possible to trip us up.

**Comment by ismena** [comment]
First update: the content isn't audio-only, t's just that we choke on combining audio streams without ever getting to text and video.
It does appear to duplicated streams, but I'm not convinced that this is out problem yet.
Will update the thread with further investigation notes.

**Comment by joeyparrish** [comment]
Fixed in 9f43011.
